rootfs/etc/nginx/lua/lua_ingress.lua

```diff
@@ -54,8 +56,16 @@ local function randomseed()
   math.randomseed(seed)
 end
 
-local function redirect_to_https()
-  return ngx.var.pass_access_scheme == "http" and (ngx.var.scheme == "http" or ngx.var.scheme == "https")
+local function redirect_to_https(location_config)
+  if location_config.force_no_ssl_redirect then
+    return false
+  end
+
+  if location_config.force_ssl_redirect or (location_config.ssl_redirect and certificate_configured_for_server(ngx.var.host)) then
+    return ngx.var.pass_access_scheme == "http" and (ngx.var.scheme == "http" or ngx.var.scheme == "https")
```

Comment:
this is a bit weird

> (ngx.var.scheme == "http" or ngx.var.scheme == "https")

can $scheme be anything other than http or https anyway? this seems like a redundant clause

Reply:
No? http://nginx.org/en/docs/http/ngx_http_core_module.html#var_scheme